File: codigos/1 - estrutura sequencial/1010 - Calculo Simples/Main.dart
import 'dart:io';

List<double> ler_entrada(){
  return stdin
          .readLineSync()!
          .split(" ")
          .sublist(1)
          .map((x) => double.parse(x))
          .toList();
}

double prod(List<double> valores){
  return valores.reduce((acc, n) => acc * n);
}

double calculo(List<double> valores1, List<double> valores2){
  return prod(valores1) + prod(valores2);
}

main(){
  List<double> valores1 = ler_entrada();
  List<double> valores2 = ler_entrada();

  String total = calculo(valores1, valores2).toStringAsFixed(2);

  print("VALOR A PAGAR: R\$ $total");
}

File: codigos/1 - estrutura sequencial/1010 - Calculo Simples/Main.clj
(defn calculo [id1 q1 v1 id2 q2 v2]
    (float (+ (* q1 v1) (* q2 v2))))

(defn main []
    (let [valores (repeatedly 6 read)]
        (println "VALOR A PAGAR: R$" (format "%.2f" (apply calculo valores)))))

(main)